frontend/app/actions/catActions.js

```diff
@@ -0,0 +1,24 @@
+import { RSAA } from 'redux-api-middleware'
+
+export const REQUEST = 'RANDOM_CAT_REQUEST'
+export const SUCCESS = 'RANDOM_CAT_SUCCESS'
+export const FAILURE = 'RANDOM_CAT_FAILURE'
+
+export function fetchRandomCat() {
+  return {
+    [RSAA]: {
+      endpoint: '/api/pet.getRandom?output=full',
+      method: 'GET',
+      types: [
+        REQUEST,
+        {
+          type: SUCCESS,
+          payload: (action, state, res) => {
+            return res.json().then(json => json['petfinder']['pet'])
```

Comment:
I think this is also where the helper function will go to clean the data of '$t' keys before the store receives it.

Reply:
You're making the assumption here that you can always JSON-ify a successful response, but as long as that's a reasonable assumption then I think you're fine.